### PR TITLE
[SPARK-28540][WEBUI] Document Environment page

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
Document Environment page

## How was this patch tested?
locally building

图片

### CODE REVIEW

File: docs/web-ui.md [hunk 1]
```diff
@@ -49,6 +49,50 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
+       title="Env tab"
+       alt="Env tab"
+       width="100%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+
+We can see that this environment page has five parts, it is a useful place to check whether your properties have
+been set correctly. The first part 'Runtime Information' simply contains the
+[runtime properties](configuration.html#runtime-environment) like versions of Java and Scala.
+The second part 'Spark Properties' lists the [application properties](configuration.html#runtime-environment) like
```
Comment (planga82) [inline docs/web-ui.md]: is this the correct link? 
configuration.html#application-properties

**Review by planga82 — COMMENTED** [review]
I think is better not put the spark version header in the images. Because we will not have change this image in spark 3.0.1...
 Don't you think?
The rest is fine for me

File: docs/web-ui.md [hunk 2]
```diff
@@ -49,6 +49,50 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
+'spark.app.name' and 'spark.driver.memory'.
+
+<p style="text-align: center;">
+  <img src="img/webui-env-hadoop.png"
+       title="Hadoop Properties"
+       alt="Hadoop Properties"
+       width="100%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+Clicking the 'Hadoop Properties' link displays properties relative to Hadoop and Yarn. Note that properties like
```
Comment (srowen) [inline docs/web-ui.md]: Yarn -> YARN
tick-quote `spark.hadoop.*`

File: docs/web-ui.md [hunk 3]
```diff
@@ -49,6 +49,50 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
 
+<p style="text-align: center;">
+  <img src="img/webui-env-tab.png"
+       title="Env tab"
+       alt="Env tab"
+       width="100%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+
+We can see that this environment page has five parts, it is a useful place to check whether your properties have
```
Comment (srowen) [inline docs/web-ui.md]: I might use a different paragraph for each of the five sections you mention.
"it is" should start a new sentence.
You can remove wording like "We can see that"

**Review by srowen — CHANGES_REQUESTED** [review]
This is probably useful just for the screenshots, but it doesn't document much else. I'm not sure there's a lot more to say though.

**Comment by srowen** [comment]
Agree about maybe keeping the version number / header out of the screenshots.

File: docs/web-ui.md (hunk 1 above)
Comment (zhengruifeng) [reply docs/web-ui.md]: good catch!

File: docs/web-ui.md [hunk 4]
```diff
@@ -107,6 +107,51 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
 
+<p style="text-align: center;">
+  <img src="img/webui-env-tab.png"
+       title="Env tab"
+       alt="Env tab"
+       width="100%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+
+This environment page has five parts, it is a useful place to check whether your properties have
```
Comment (srowen) [inline docs/web-ui.md]: Nit: "parts, it" -> "parts. It"

File: docs/web-ui.md [hunk 5]
```diff
@@ -107,6 +107,51 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
+been set correctly.
+The first part 'Runtime Information' simply contains the [runtime properties](configuration.html#runtime-environment)
+like versions of Java and Scala.
+The second part 'Spark Properties' lists the [application properties](configuration.html#application-properties) like
+'spark.app.name' and 'spark.driver.memory'.
```
Comment (srowen) [inline docs/web-ui.md]: You could back-tick-quote these, but it's not a big deal.

Comment (zhengruifeng) [reply docs/web-ui.md]: Do you mean adding links for 'spark.app.name' and 'spark.driver.memory'?
It seems that there isn't an anchor link for a specific property in `https://spark.apache.org/docs/latest/configuration.html#application-properties`.

Comment (srowen) [reply docs/web-ui.md]: No I just mean like `spark.app.name`

File: docs/web-ui.md [hunk 6]
```diff
@@ -107,6 +107,51 @@ sizes and using executors for all partitions in an RDD or DataFrame.
...
+
+<p style="text-align: center;">
+  <img src="img/webui-env-class.png"
+       title="Classpath Entries"
+       alt="Classpath Entries"
+       width="100%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+
+The last part 'Classpath Entries' illustrates the classes loaded from different sources, which is very useful
```
Comment (srowen) [inline docs/web-ui.md]: Really minor but I might say "lists" or "shows" rather than "illustrates"

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**